### PR TITLE
Move detect system to its own module

### PR DESCRIPTION
**Pull request by elfosardo**
Moving more content from the detect script to specific module to
separate parsing logic.

### CODE REVIEW

File: hardware/detect.py [hunk 1]
```diff
@@ -480,8 +276,10 @@ def main():
     hrdw.append(megacli.detect())
     hrdw.append(diskinfo.detect())
 
-    if not detect_system(hrdw):
+    if not system.detect():
         sys.exit(1)
+    hrdw.append(system.detect())
```
Comment (ErwanAliasr1) [inline hardware/detect.py:281]: This makes a double detection, couldn't we have a single ?

File: hardware/system.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                    nic_id = nic_id.replace(':', '')
+
+                detect_utils.get_ethtool_status(hw_lst, name.text)
+                detect_utils.get_lld_status(hw_lst, name.text)
+
+        detect_utils.fix_bad_serial(hw_lst, uuid, mobo_id, nic_id)
+
+    else:
+        sys.stderr.write("Unable to run lshw: %s\n" % output)
+        return False
```
Comment (ErwanAliasr1) [inline hardware/system.py:198]: We should probably return [] here to stay consistent about the function output

**Review by ErwanAliasr1 — CHANGES_REQUESTED** [review]

File: hardware/detect.py (hunk 1 above)
Comment (elfosardo) [reply hardware/detect.py:281]: yeah, we could assign the result to a variable and check that

File: hardware/system.py (hunk 2 above)
Comment (elfosardo) [reply hardware/system.py:198]: whoops I didn't see this, thanks!

**Review by ErwanAliasr1 — APPROVED** [review]

**Pull request merged**